shadowsocks/tcp.go

```diff
@@ -149,7 +150,7 @@ type TCPService interface {
 }
 
 // proxyConnection will route the clientConn according to the address read from the connection.
-func proxyConnection(clientConn onet.DuplexConn, proxyMetrics *metrics.ProxyMetrics, allowAllIPs bool) *onet.ConnectionError {
+func proxyConnection(clientConn onet.DuplexConn, proxyMetrics *metrics.ProxyMetrics, ipPolicy func(net.IP) *onet.ConnectionError) *onet.ConnectionError {
```

Comment:
Please rename ipPolicy so it's a verb, rather than a noun. Maybe something like `(verify|ensure|assert)AllowedIP()`?

Reply:
renamed to `checkIP`.

Reply:
For readability, let's express what you are checking. Are you checking if it's malformed? Public? In this case, we are checking if the IP is allowed. `checkAllowedIP()` would be clearer.

Reply:
Done